src/sql/workbench/parts/objectExplorer/browser/connectionTreeAction.ts

```diff
@@ -41,40 +41,37 @@ export class RefreshAction extends Action {
 		super(id, label);
 		this._tree = tree;
 	}
-	public run(): Promise<boolean> {
+	public async run(): Promise<boolean> {
 		let treeNode: TreeNode;
 		if (this.element instanceof ConnectionProfile) {
 			let connection: ConnectionProfile = this.element;
 			if (this._connectionManagementService.isConnected(undefined, connection)) {
 				treeNode = this._objectExplorerService.getObjectExplorerNode(connection);
 				if (treeNode === undefined) {
-					this._objectExplorerService.updateObjectExplorerNodes(connection.toIConnectionProfile()).then(() => {
-						treeNode = this._objectExplorerService.getObjectExplorerNode(connection);
-					});
+					await this._objectExplorerService.updateObjectExplorerNodes(connection.toIConnectionProfile());
+					treeNode = this._objectExplorerService.getObjectExplorerNode(connection);
 				}
 			}
 		} else if (this.element instanceof TreeNode) {
 			treeNode = this.element;
 		}
 
 		if (treeNode) {
-			return this._tree.collapse(this.element).then(() => {
-				return this._objectExplorerService.refreshTreeNode(treeNode.getSession(), treeNode).then(() => {
-
-					return this._tree.refresh(this.element).then(() => {
-						return this._tree.expand(this.element);
-					}, refreshError => {
-						return Promise.resolve(true);
-					});
-				}, error => {
+			try {
+				await this._tree.collapse(this.element);
+				try {
+					await this._objectExplorerService.refreshTreeNode(treeNode.getSession(), treeNode);
+				} catch (error) {
 					this.showError(error);
 					return Promise.resolve(true);
-				});
-			}, collapseError => {
-				return Promise.resolve(true);
-			});
+				}
+				await this._tree.refresh(this.element);
+				return this._tree.expand(this.element);
+			} catch (ex) {
+				return true;
```

Comment:
Hm, might be a good idea to log the error at least. I would think this wouldn't be an expected scenario and silently ignoring it might be hiding other issues